scripts/cleanup_versions.py

```diff
@@ -0,0 +1,117 @@
+import sys
+import boto3
+
+
+class CleanupVersions:
+    def __init__(self):
+        self.lambda_client = boto3.client("lambda")
+        self.appconfig_client = boto3.client("appconfig")
+        self.environment = sys.argv[1]
+
+    def start(self):
+        self.delete_hosted_configuration_versions()
+        self.delete_lambda_layer_versions()
+
+    def get_app_config_application_id(self) -> str:
+        current_applications = self.appconfig_client.list_applications()
+        for app in current_applications["Items"]:
+            if f"-{self.environment}" in app["Name"]:
+                return app["Id"]
+
+    def get_app_config_profile_id(self, application_id: str) -> str:
+        config_profiles = self.appconfig_client.list_configuration_profiles(
+            ApplicationId=application_id
+        )
+        for profile in config_profiles["Items"]:
+            if f"-{self.environment}" in profile["Name"]:
+                return profile["Id"]
+
+    def get_hosted_configuration_versions(self):
+        print(f"\nGathering AppConfig hosted configuration versions on {self.environment}...")
+        application_id = self.get_app_config_application_id()
+        config_profile_id = self.get_app_config_profile_id(application_id)
+
+        current_hosted_configuration_versions = self.appconfig_client.list_hosted_configuration_versions(
+            ApplicationId=application_id,
+            ConfigurationProfileId=config_profile_id
+        )
+
+        return current_hosted_configuration_versions["Items"]
+
+    def delete_hosted_configuration_versions(self):
+        excess_hosted_config_versions = self.get_hosted_configuration_versions()
```

Comment:
I'm probably missing something, but how does it exclude the original version?

Reply:
No you haven't missed anything good spot, but it was done on purpose. It doesn't look like the TF state cares about tracking the hosted configuration versions as they're not something that are really a usable feature in terraform (which is why we're in this predicament in the first place), and so removing every hosted configuration version doesn't cause any issues when it comes to the terraform destroy. We'd just be adding unnecessary filtering logic to remove the original version